### ISSUE TITLE
Lint rule react-rules-of-hooks misidentifies a conditional hook

### ISSUE THREAD

**Issue by paul-sachs** [denoland/deno_lint#1426]
When using the new lint rules, deno falsely identifies the `useState` as conditional.

Example can be seen on this repo [here](https://github.com/paul-sachs/deno-react/blob/psachs/linting-bug/client/src/App.tsx#L43)

### Lint Name

react-rules-of-hooks

### Code Snippet

```tsx
const Comp = () => {
  useDocumentTitle(`Subscription · ${owner ?? ""}`);

  const [count, setCount] = useState(0);

  return null
}
```

### Expected Result

No error

### Actual Result

Identifies `useState` as conditional.

### Version

deno 2.2.2 (stable, release, aarch64-apple-darwin)
v8 13.4.114.9-rusty
typescript 5.7.3
